### ISSUE TITLE
Editor enhancement: Preview of complete file in sidebar

### ISSUE THREAD

**Issue by JakobJakobson** [spyder-ide/spyder#6448]
Dear developers,

could you implement a preview of the complete file that you are currently working on in a sidebar such as Kate/KWrite/GEdit have implemented it. (I attached screenshots of Kate and GEdit.)

Regards

Jakob

![kate_sidebar_filepreview](https://user-images.githubusercontent.com/31574479/36154617-4ccf7f2e-10d2-11e8-9ac3-f7e7597271fd.png)
![gedit_sidebar_filepreview](https://user-images.githubusercontent.com/31574479/36154623-506f6130-10d2-11e8-948f-c28ef7ffe95d.png)

**Comment by ccordoba12** [comment]
This is called a minimap, and it seems quite easy to implement:

https://www.youtube.com/watch?v=FHl-5Jc0w5o

Would you like to give it a chance? It seems a fun and not so hard project to work on.

**Comment by JakobJakobson** [comment]
I'll try to get to it at the weekend, let's see how far I'll get

**Comment by ccordoba12** [comment]
Great! Please do your PR against master and take a look at the `ScrollFlagArea` panel

https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/panels/scrollflag.py

I think it has most of what you need, right @jnsebgosselin?

**Comment by jnsebgosselin** [comment]
Yes, the structure to do it is all there. You'll only need to think of a way to produce the preview. I would start by looking at the `TextEditBaseWidget`.

https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/sourcecode/base.py#L230

**Comment by jnsebgosselin** [comment]
@ccordoba12 With all the new panels we are adding in Spyder 4 (codefolding, debugger panel, classes and functions selector, this new file preview panel), I think we will need to add in the preferences a dedicated section where users can show/hide the panels that they want or do not want in the editor.

![image](https://user-images.githubusercontent.com/10170372/36172370-aa337886-10d3-11e8-9c18-a3deac78af7d.png)

**Comment by ccordoba12** [comment]
@jnsebgosselin, agreed. Could you open a new issue for that? Thanks!
